interp/interp.go

```diff
@@ -0,0 +1,158 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package interp
+
+import (
+	"sort"
+)
+
+// Predictor predicts the value of a function. It handles both
+// interpolation and extrapolation.
+type Predictor interface {
+	// Predict returns the predicted value at x.
+	Predict(float64) float64
```

Comment:
Perhaps directly extend this to take another parameter specifying the order of the derivative returned?

Reply:
I wouldn't do it right now.